src/OpenTelemetry.Exporter.Prometheus.HttpListener/Internal/PrometheusSerializer.cs

```diff
@@ -234,6 +234,7 @@ public static int WriteMetricName(byte[] buffer, int cursor, PrometheusMetric me
     {
         // Metric name has already been escaped.
         var name = openMetricsRequested ? metric.OpenMetricsName : metric.Name;
+        Guard.ThrowIfNullOrWhitespace(name, nameof(name));
```

Comment:
I like that you added the check here, but this Guard method is going to throw an `ArgumentException`. This will cause confusion because this method doesn't have a "name" argument.

Reply:
I switched these to `Debug.Assert`s. Typically in our internal code we prefer `Debug.Assert` over `Guard.*` because the former gets removed in release builds.